### PR TITLE
Remove Provider. Long live Riverpod

### PR DESCRIPTION
**Pull request by bizz84**
- [x] Port main codebase to Riverpod
- [x] Check all `StreamProvider` errors
- [ ] Remove logger if no longer needed
- [x] Update tests
- [ ] Fix integration tests
- [x] Update Readme

### CODE REVIEW

**Comment by bizz84** [comment]
Stuck here:
https://github.com/rrousselGit/river_pod/issues/144

File: lib/app/auth_widget.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    @required this.signedInBuilder,
+    @required this.nonSignedInBuilder,
+  }) : super(key: key);
+  final WidgetBuilder nonSignedInBuilder;
+  final WidgetBuilder signedInBuilder;
+
+  @override
+  Widget build(BuildContext context, ScopedReader watch) {
+    final firebaseAuth = watch(firebaseAuthProvider);
+    final authStateChangesProvider =
```
Comment (rrousselGit) [inline lib/app/auth_widget.dart:19]: Don't do that. The provider should be static/global.

File: lib/app/providers.dart [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:logger/logger.dart';
+import 'package:starter_architecture_flutter_firebase/services/firestore_database.dart';
+
+final firebaseAuthProvider =
+    Provider<FirebaseAuth>((ref) => FirebaseAuth.instance);
+
+final databaseProvider =
+    ScopedProvider<FirestoreDatabase>((ref) => throw UnimplementedError());
```
Comment (rrousselGit) [inline lib/app/providers.dart:10]: tip:

There's a shorthand for that:

```dart
final databaseProvider = ScopedProvider<FirestoreDatabase>(null);
```

Comment (rrousselGit) [reply lib/app/providers.dart:10]: But considering what you are doing in `AuthWidget`, you may want to remove the AuthWidget from `ScopedProvider` and instead do:

```dart
final databaseProvider = Provider<FirestoreDatabase>((ref) {
  final auth = ref.watch(authStateChangesProvider);

  if (auth.data?.value?.user?.id != null) {
    return FirestoreDatabase(auth.data.value.user.id);
  }
  return null;  
});
```

Comment (bizz84) [reply lib/app/providers.dart]: Thanks for the suggestions. I fixed all the providers and it works much better now :)

File: lib/app/auth_widget.dart [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    return authStateChanges.when(
+      data: (user) => _data(context, user),
+      loading: () => const Scaffold(
+        body: Center(
+          child: CircularProgressIndicator(),
+        ),
+      ),
+      error: (_, __) => const Scaffold(
+        body: Center(
+          child: Text('Error reading authStateChanges()'),
```
Comment (bizz84) [inline lib/app/auth_widget.dart:27]: TODO: Update

**Pull request merged**